File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string>ans;
        
        for(int i=0;i<s.size();)
        {
            string temp="";
            
            while(s[i]!=' ' && i<s.size())
            {
                
                // cout<<s[i];
                temp=temp+s[i];
                i++;
            }
            // cout<<temp<<" ";
            i++;
            if(temp.size()!=0)
            ans.push_back(temp);
        }
        string res="";
        for(int i=ans.size()-1;i>=0;i--)
        {
            for(int j=0;j<ans[i].size();j++)
            {
              res+=ans[i][j];  
            }
            if(i!=0)
            res+=' ';
        }
        return res;
        
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        
        boolean flag =false;
        StringBuilder ans = new StringBuilder();
        for(int i=s.length()-1 ;i>=0;i--){
            StringBuilder element = new StringBuilder();
            while(i>=0  && s.charAt(i)!=' '){
                element.insert(0,s.charAt(i));
                
                i--;
            }
            if(!element.isEmpty()){
                if(flag == true){
                    ans.append(" ");
                }
                ans.append(element);
                flag =true;
            }
        }
        return ans.toString();

        
    }
}